### PR TITLE
Remove tower from axum's public API

### PR DESCRIPTION
**Pull request by davidpdrsn**
Instead rely on `tower-service` and `tower-layer`. `tower` itself is
only used internally.

Fixes https://github.com/tokio-rs/axum/issues/186

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -18,7 +18,8 @@ members = ["examples/*"]
 default = ["tower-log"]
 ws = ["tokio-tungstenite", "sha-1", "base64"]
 multipart = ["multer", "mime"]
-tower-log = ["tower/log"]
+# tower-log = ["tower/log"]
+tower-log = []
```
Comment (jplatte) [inline Cargo.toml:22]: Is this change intentional?

File: src/routing/future.rs [hunk 2]
```diff
@@ -179,3 +181,9 @@ where
         self.project().inner.poll(cx)
     }
 }
+
+opaque_future! {
+    /// Response future from [`MakeRouteService`] services.
+    pub type MakeRouteServiceFuture<S> =
+        futures_util::future::Ready<Result<S, Infallible>>;
```
Comment (jplatte) [inline src/routing/future.rs:188]: With axum's current MSRV you could use the `std` equivalent (introduced in 1.48.0). Or does that not work due to dependencies using this type?

File: Cargo.toml (hunk 1 above)
Comment (davidpdrsn) [reply Cargo.toml:22]: Nope 😊 going through everything now making sure its good.

File: src/routing/future.rs (hunk 2 above)
Comment (davidpdrsn) [reply src/routing/future.rs:188]: Whoa I did not realize `std::future::Ready` was stable 🎊 Gonna try using that instead!

File: Cargo.toml (hunk 1 above)
Comment (davidpdrsn) [reply Cargo.toml]: Done

File: src/routing/future.rs (hunk 2 above)
Comment (davidpdrsn) [reply src/routing/future.rs:188]: I'll address this in a follow up PR.
